libtenzir/builtins/functions/string.cpp

```diff
@@ -197,6 +197,157 @@ class nullary_method : public virtual method_plugin {
+        auto result_type = string_type{};
+        auto result_arrow_type
+          = std::shared_ptr<arrow::DataType>{result_type.to_arrow_type()};
+        auto subject = eval(subject_expr);
+        auto f = detail::overload{
+          [&](const arrow::StringArray& array) {
+            auto max = max_replacements ? max_replacements->inner : -1;
+            auto options = arrow::compute::ReplaceSubstringOptions(
+              pattern, replacement, max);
+            auto result = arrow::compute::CallFunction("replace_substring",
+                                                       {array}, &options);
+            if (not result.ok()) {
+              diagnostic::warning("{}", result.status().ToString())
+                .primary(subject_expr)
+                .emit(ctx);
+              return series::null(result_type, subject.length());
+            }
+            if (not result->type()->Equals(result_arrow_type)) {
+              result = arrow::compute::Cast(result.MoveValueUnsafe(),
+                                            result_arrow_type);
+              TENZIR_ASSERT(result.ok(), result.status().ToString());
+            }
+            return series{result_type, result.MoveValueUnsafe().make_array()};
+          },
+          [&](const arrow::NullArray& array) {
+            return series::null(result_type, array.length());
+          },
+          [&](const auto&) {
+            diagnostic::warning("`{}` expected `string`, but got `{}`", name(),
+                                subject.type.kind())
+              .primary(subject_expr)
+              .emit(ctx);
+            return series::null(result_type, subject.length());
+          },
+        };
+        return caf::visit(f, *subject.array);
+      });
+  }
+};
+
+class slice : public virtual method_plugin {
+public:
+  auto name() const -> std::string override {
+    return "slice";
+  }
+
+  auto make_function(invocation inv, session ctx) const
+    -> failure_or<function_ptr> override {
+    auto subject_expr = ast::expression{};
+    auto start = located<int64_t>{};
+    auto stop = std::optional<located<int64_t>>{};
+    auto step = std::optional<located<int64_t>>{};
+    TRY(argument_parser2::method(name())
+          .add(subject_expr, "<string>")
+          .add(start, "<start>")
+          .add("stop", stop)
+          .add("step", step)
+          .parse(inv, ctx));
+    if (start.inner < 0) {
+      diagnostic::error("`start` must be at least 0, but got {}", start.inner)
+        .primary(start)
+        .emit(ctx);
+    }
+    if (stop) {
+      if (stop->inner < 0) {
+        diagnostic::error("`stop` must be at least 0, but got {}", stop->inner)
+          .primary(*stop)
+          .emit(ctx);
+      }
+    }
+    if (step) {
+      if (step->inner <= 0) {
+        diagnostic::error("`step` must be greater 0, but got {}", step->inner)
+          .primary(*step)
+          .emit(ctx);
+      }
+    }
```

Comment:
`some_string.slice(-2)` could also be useful to get the last two characters/grapheme clusters. Could make this behavior the same as what we have for the `slice` operator.

Reply:
Turns out that was already supported by the Arrow function. I added a test and made it possible.